src/Comparator.php

```diff
@@ -0,0 +1,58 @@
+<?php
+declare(strict_types=1);
+
+namespace Roave\ApiCompare;
+
+use Roave\BetterReflection\Reflector\ClassReflector;
+use Roave\BetterReflection\Reflector\Exception\IdentifierNotFound;
+
+final class Comparator
+{
+    public function compare(ClassReflector $oldApi, ClassReflector $newApi): array
+    {
+        $changelog = [];
+
+        foreach ($oldApi->getAllClasses() as $oldClass) {
+            try {
+                $newClass = $newApi->reflect($oldClass->getName());
+            } catch (IdentifierNotFound $exception) {
+                $changelog[] = sprintf('[BC] Class %s has been deleted', $oldClass->getName());
+                continue;
+            }
+
+            foreach ($oldClass->getMethods() as $oldMethod) {
+                // @todo ignore private methods
+                try {
+                    $newMethod = $newClass->getMethod($oldMethod->getName());
+                } catch (\OutOfBoundsException $exception) {
+                    $changelog[] = sprintf(
+                        '[BC] Method %s in class %s has been deleted',
+                        $oldMethod->getName(),
+                        $oldClass->getName()
+                    );
+                    continue;
+                }
+
+                foreach ($oldMethod->getParameters() as $oldParameter) {
+                    $newParameter = $newMethod->getParameter($oldParameter->getName());
```

Comment:
The parameter should not be fetched by name, but by position. A simple test of a BC compliant change is:

before:
```php
function foo(int $a, string $b) {}
```

after:
```php
function foo(int $b, string $a) {}
```

Also, a rename per-se does not cause a BC break:


before:
```php
function foo(int $a, string $b) {}
```

after:
```php
function foo(int $a, string $c) {}
```